packages/next-swc/crates/next-core/src/next_app/app_page_entry.rs

```diff
@@ -127,30 +136,68 @@ pub async fn get_app_page_entry(
     .cell())
 }
 
-async fn wrap_edge_entry(
-    context: Vc<ModuleAssetContext>,
+#[turbo_tasks::function]
+async fn wrap_edge_page(
+    context: Vc<Box<dyn AssetContext>>,
     project_root: Vc<FileSystemPath>,
     entry: Vc<Box<dyn Module>>,
+    page: AppPage,
+    next_config: Vc<NextConfig>,
 ) -> Result<Vc<Box<dyn Module>>> {
-    const INNER: &str = "INNER_RSC_ENTRY";
+    const INNER: &str = "INNER_PAGE_ENTRY";
+
+    let next_config = &*next_config.await?;
+
+    // TODO(WEB-1824): add build support
+    let build_id = "development";
+    let dev = true;
+
+    // TODO(timneutkens): remove this
+    let is_server_component = true;
+
+    let server_actions_body_size_limit = &next_config.experimental.server_actions_body_size_limit;
+    let sri_enabled = !dev
+        && next_config
+            .experimental
+            .sri
+            .as_ref()
+            .map(|sri| sri.algorithm.as_ref())
+            .is_some();
 
     let source = load_next_js_template(
-        "edge-app-route.js",
+        "edge-ssr-app.js",
         project_root,
         indexmap! {
             "VAR_USERLAND" => INNER.to_string(),
+            "VAR_PAGE" => page.to_string(),
+            "VAR_BUILD_ID" => build_id.to_string(),
+        },
+        indexmap! {
+            "sriEnabled" => serde_json::Value::Bool(sri_enabled).to_string(),
+            "nextConfig" => serde_json::to_string(next_config)?,
+            "isServerComponent" => serde_json::Value::Bool(is_server_component).to_string(),
+            "dev" => serde_json::Value::Bool(dev).to_string(),
+            "serverActionsBodySizeLimit" => serde_json::to_string(server_actions_body_size_limit)?,
+        },
+        indexmap! {
+            "incrementalCacheHandler" => None,
         },
-        indexmap! {},
-        indexmap! {},
     )
     .await?;
 
     let inner_assets = indexmap! {
         INNER.to_string() => entry
     };
 
-    Ok(context.process(
+    let wrapped = context.process(
```

Comment:
Is `wrapped` a good name, since it's not yet wrapped?

Reply:
kinda already wrapped by the edge-ssr-app template above 👀